ca/certificates.go

```diff
@@ -323,6 +323,42 @@ func (rca *RootCA) ParseValidateAndSignCSR(csrBytes []byte, cn, ou, org string)
 	return cert, nil
 }
 
+// CrossSignCACertificate creats a CSR from a root certificate and signs it using the current root signer
+func (rca *RootCA) CrossSignCACertificate(otherRootCert []byte) ([]byte, error) {
+	if !rca.CanSign() {
+		return nil, ErrNoValidSigner
+	}
+
+	newCert, err := helpers.ParseCertificatePEM(otherRootCert)
+	if err != nil {
+		return nil, err
```

Comment:
Let's wrap this error with a mention that it's a parse error of the supplied certificate.